connector/routingconnector/router.go

```diff
@@ -81,80 +80,121 @@ type routingItem[C any] struct {
 	statementContext   string
 }
 
-func (r *router[C]) buildParsers(table []RoutingTableItem, settings component.TelemetrySettings) error {
-	var buildResource, buildSpan, buildMetric, buildDataPoint, buildLog bool
-	for _, item := range table {
-		switch item.Context {
-		case "", "resource":
-			buildResource = true
-		case "span":
-			buildSpan = true
-		case "metric":
-			buildMetric = true
-		case "datapoint":
-			buildDataPoint = true
-		case "log":
-			buildLog = true
-		}
+func (r *router[C]) buildParsers(_ []RoutingTableItem, settings component.TelemetrySettings) error {
+	// Context inference priority list: when a condition uses an ambiguous path (one that exists
+	// in multiple contexts), the inferrer tries each context in order until one can parse it.
+	//
+	// "resource" is first for backward compatibility: before context inference existed, the
+	// routing connector only supported resource context. Existing configs with conditions like
+	// `attributes["env"] == "prod"` must continue to resolve to resource.attributes.
+	//
+	// The remaining order matters less in practice because most ambiguous paths (like "attributes")
+	// exist in resource anyway, and non-ambiguous paths (like "body", "severity_text", "name")
+	// only exist in one context regardless of priority. That said, the order is sorted based on which
+	// events are most common in practice, hence 'span' is first.
+	priorities := []string{
```

Comment:
@edmocosta do you have any advice for how to handle potential new types that get added to OTel in the future? I'm not nuts about this list sitting here, but it's no different than the previous case statements I suppose.